adb_graphics/datahandler/gribdata.py

```diff
@@ -481,128 +281,185 @@ class fieldData(UPPData):
+
+    def field_column_max(self, values: DataArray, **kwargs):  # noqa: ARG002
+        """Returns the column max of the values."""
+
+        return values.max(dim=self.vertical_coord)
 
-    def fire_weather_index(self, values, **kwargs):
+    def field_diff(self, values: DataArray, variable2: str, level2: str, **kwargs):
+        """Subtracts the values from variable2 from self.field."""
 
-        # pylint: disable=unused-argument
+        value2 = self.values(
+            name=variable2, level=level2, do_transform=kwargs.get("do_transform", True)
+        )
+        diff = values - value2
+        value2.close()
+
+        return diff
 
-        '''
-        Generates a field of Fire Weather Index
+    def field_mean(
+        self,
+        values: DataArray,
+        levels: list,
+        **kwargs,
+    ):
+        """Returns the mean of the values over the vertical dimension."""
+
+        levels = kwargs["global_levels"] if "global" in self.model else levels
+        levs = [int(x[:-2]) for x in levels]
+        return values.sel(isobaricInhPa=levs).mean("isobaricInhPa")
+
+    def field_sum(self, values: DataArray, variable2: str, level2: str, **kwargs):
+        """Return the sum of the values."""
+
+        value2 = self.values(
+            name=variable2, level=level2, do_transform=kwargs.get("do_transform", True)
+        )
+        sum2 = values + value2
+        value2.close()
+
+        return sum2
+
+    def fire_weather_index(self, values: DataArray, **kwargs):  # noqa: ARG002
+        """
+        Generates a field of Fire Weather Index.
 
         This method uses wrfprs data to find regions where
         weather conditions are most likely to lead to wildfires.
 
-        '''
+        """
 
         # Gather fields from the input
-        veg = values # Chose this value as the main one in the default_specs
-        temp = self.values(name='temp', level='2m', do_transform=False)
-        dewpt = self.values(name='dewp', level='2m', do_transform=False)
-        weasd = self.values(name='weasd', level='sfc', do_transform=False)
-        gust = self.values(name='gust', level='10m', do_transform=False)
-        soilm = self.values(name='soilm', level='sfc', do_transform=False)
+        veg = values
+
+        temp = self.values(level="2m", name="temp", do_transform=False)
+        dewpt = self.values(level="2m", name="dewp", do_transform=False)
+        weasd = self.values(level="sfc", name="weasd", do_transform=False)
+        gust = self.values(level="10m", name="gust", do_transform=False)
+        soilm = self.values(level="sfc", name="soilm", do_transform=False)
 
         # A few derived fields
         dewpt_depression = temp - dewpt
-        dewpt_depression = np.where(dewpt_depression < 0, 0, dewpt_depression)
-        dewpt_depression = np.maximum(15.0, dewpt_depression)
+        dewpt_depression = where(dewpt_depression < 0, 0, dewpt_depression)
+        dewpt_depression = ufuncs.maximum(15.0, dewpt_depression)
```

Comment:
Using the xarray-provided function here since these are DataArrays. Sometimes that gave some significant speedup, so adopted it where I could.